### DISCUSSION TITLE
Does Lennox have an API for uploading a screensaver photo?

### DISCUSSION THREAD

**Discussion by 42Network** [PeteRager/lennoxs30#279] [Ideas]
Lennox provides a way to upload a photo using the mobile app to be used as a "screensaver" on the thermostat. Can this be done using an API? If so, I was thinking it would be interesting to dynamically generate a "photo" with other HA data, such as historical temperature trends, historical blower CFM, or literally any other HA data. This way the thermostat idle screen could be used as a read-only HA display. Thoughts?

**Comment by PeteRager** [comment]
That's certainly an interesting idea.  Mine is right next to the front door - I could see having a list of notifications or reminders could be useful.  I'll see if I can find the API and create a curl command so we can test it.  My guess is we'd need to provide a jpg or png file.   

Do you know if there is a way to render a Lovelace view into a JPG and save it to disk?

**Reply by 42Network** [reply]
A bit of googling found this active project that is meant to provide a rendered JPG of a Lovelace dashboard as a Kindle e-reader screensaver. The project is recently updated and may not take much modification to render in color rather than the current B&W. It appears to run as a Docker container which you can query on port 5000 for the most recent rendering of the configured dashboard. 

Too bad the Lennox thermostat can't be configured to point to a URL directly for the "screensaver". But if you find there is an API to push an image to the thermostat, implementing this read-only HA dashboard could be as simple as a scheduled task to read from port 5000 of the Docker container and then call the REST API to load into the thermostat every X minutes.

https://github.com/sibbl/hass-lovelace-kindle-screensaver

**Comment by PeteRager** [comment]
Here are some of the requirements:

    private static final int S30_HORIZONTAL_DIMENSION = 640;
    private static final int S30_MAX_FILENAME_CHARS = 64;
    private static final int S30_MAX_SIZE_BYTES = 20971520;
    private static final int S30_TSTAT_HORIZONTAL_DIMENSION = 1280;
    private static final int S30_TSTAT_VERTICAL_DIMENSION = 800;
    private static final int S30_VERTICAL_DIMENSION = 400;
    private static final String[] SUPPORTED_IMAGE_FILE_TYPES = {"jpeg", "jpg", "png", "bmp", "gif", "heic"};

The Android code to upload photos is a thousand lines, so I don't think we'll be able to figure it out from that.  We'll need to capture the transaction through a burp proxy.
